File: Random/1.Find Pivot Index.py
#    Author : Mohamed Yousef 
#    Date   : 2022-12-07
import sys,math,bisect,collections,itertools,heapq
from collections import defaultdict,deque
def solve(nums):
    n=len(nums)
    sum1=0
    sum2=sum(nums[1:])
    i=0
    for i in range(1,n):
        if sum1==sum2:
            return i-1
        # print(sum1,sum2)
        sum1+=nums[i-1]
        sum2-=nums[i]
    if sum1==sum2:
        return i
    return -1
print(solve([1,7,3,6,5,6])) 

File: Random/1.Find Pivot Index.js
/*
    Author : Mohamed Yousef 
    Date   : 2022-12-07
*/
let solve=function(nums){
    let n=nums.length;
    let sum1=0
    // let i=0
    s=nums.slice(1,n)
    if (n==1) return 0
    let sum2=s.reduce((ele1,ele2)=>ele1+ele2)
    for(var i=1;i<n;i++){
        if (sum1==sum2){
            return i-1
        }
        console.log(`${sum1} ${sum2}`)
        sum1+=nums[i-1]
        sum2-=nums[i]
    }
    if (sum1==sum2) return i-1
    return -1
}
console.log(solve([1,0]));
